java/java.source.base/src/org/netbeans/api/java/source/SourceUtils.java

```diff
@@ -628,7 +630,9 @@ private static FileObject findSourceForBinary(FileObject binaryRoot, FileObject
     private static FileObject findMatchingChild(String sourceFileName, Collection<FileObject> folders, boolean caseSensitive) {
         final Match matchSet = caseSensitive ? new CaseSensitiveMatch(sourceFileName) : new CaseInsensitiveMatch(sourceFileName);
         for (FileObject folder : folders) {
-            for (FileObject child : folder.getChildren()) {
+            FileObject[] children = folder.getChildren();
+            Arrays.sort(children, Comparator.comparing(FileObject::getNameExt)); // for determinism
```

Comment:
Otherwise test may randomly pass or fail depending on filesystem iteration order. Would not affect production use since a `*-sources.jar` would return elements in ZIP order (typically folders first).